src/AspNetCoreModuleV2/InProcessRequestHandler/inprocess/inprocessapplication.cpp

```diff
@@ -658,10 +374,30 @@ IN_PROCESS_APPLICATION::LoadManagedApplication
     }
 
     // Set up stdout redirect
-    SetStdOut();
 
+    if (m_pLoggerProvider == NULL)
```

Comment:
You are already taking the lock unconditionally in this method, doing double check lock doesn't have any benefits here.

Reply:
Yeah it does. This check turns into:

```c++
if (m_pLoggerProvider == NULL)
    {
        AcquireLockExclusive(m_srwLock);
        if (m_pLoggerProvider == NULL)
        {
            m_pLoggerProvider = LoggingHelpers::CreateLoggingProvider(
                m_pConfig->QueryStdoutLogEnabled(),
                GetConsoleWindow(),
                m_pConfig->QueryStdoutLogFile()->QueryStr(),
                m_pConfig->QueryApplicationPhysicalPath()->QueryStr());

            m_pLoggerProvider->Start();
        }
        ReleaseLockExclusive(m_srwLock);
    }
```
After acquiring the lock, if m_pLoggerProvider was set, we don't want to re-set it.

Reply:
You are acquiring lock in the next block unconditionally just put:
```
if (m_pLoggerProvider == NULL)
        {
            m_pLoggerProvider = LoggingHelpers::CreateLoggingProvider(
                m_pConfig->QueryStdoutLogEnabled(),
                GetConsoleWindow(),
                m_pConfig->QueryStdoutLogFile()->QueryStr(),
                m_pConfig->QueryApplicationPhysicalPath()->QueryStr());

            m_pLoggerProvider->Start();
        }

```

under the next lock

Reply:
```

    {
        SRWLockWrapper lock(m_srwLock);
      
        if (m_pLoggerProvider == NULL)
        {
            m_pLoggerProvider = LoggingHelpers::CreateLoggingProvider(
                m_pConfig->QueryStdoutLogEnabled(),
                GetConsoleWindow(),
                m_pConfig->QueryStdoutLogFile()->QueryStr(),
                m_pConfig->QueryApplicationPhysicalPath()->QueryStr());
            if (m_pLoggerProvider == NULL)
            {
                hr = E_OUTOFMEMORY;
                goto Finished;
            }

            m_pLoggerProvider->Start();
        }

        if (m_status != APPLICATION_STATUS::STARTING)
        {
            if (m_status == APPLICATION_STATUS::FAIL)
            {
                hr = E_APPLICATION_ACTIVATION_EXEC_FAILURE;
            }
            else if (m_status == APPLICATION_STATUS::SHUTDOWN)
            {
                hr = HRESULT_FROM_WIN32(ERROR_SHUTDOWN_IS_SCHEDULED);
            }

            goto Finished;
        }
        m_hThread = CreateThread(
NULL, // default security attributes


```